src/main/java/com/nha/abdm/wrapper/hip/hrp/consent/ConsentService.java

```diff
@@ -1,24 +1,121 @@
+    this.requestManager = requestManager;
+    this.hipClient = hipClient;
+  }
+
+  /**
+   * The callback from ABDM gateway after consentGrant for dataTransfer, POST method for /on-notify
+   *
+   * @param hipNotifyRequest careContext and demographics details are provided, and implement a
+   *     logic to check the existence of the careContexts.
+   */
+  public void notifyOnReceived(HIPNotifyRequest hipNotifyRequest) throws IllegalDataStateException {
+    HIPOnNotifyRequest hipOnNotifyRequest = null;
+    if (hipNotifyRequest != null
+        && hipNotifyRequest.getNotification() != null
+        && hipNotifyRequest.getNotification().getConsentDetail() != null
+        && hipNotifyRequest.getNotification().getConsentDetail().getPatient() != null) {
+      HIPNotification hipNotification = hipNotifyRequest.getNotification();
+
+      RespRequest responseRequestId =
+          RespRequest.builder().requestId(hipNotifyRequest.getRequestId()).build();
+
+      if (patientService.isCareContextPresent(
+          hipNotification.getConsentDetail().getCareContexts())) {
+        Consent consent =
+            Consent.builder()
+                .status(hipNotification.getStatus())
+                .consentDetail(hipNotification.getConsentDetail())
+                .signature(hipNotification.getSignature())
+                .build();
+        patientService.addConsent(hipNotification.getConsentDetail().getPatient().getId(), consent);
+        ConsentAcknowledgement dataAcknowledgement =
+            ConsentAcknowledgement.builder()
+                .status("OK")
+                .consentId(hipNotifyRequest.getNotification().getConsentId())
+                .build();
+        hipOnNotifyRequest =
+            HIPOnNotifyRequest.builder()
+                .requestId(UUID.randomUUID().toString())
+                .timestamp(Utils.getCurrentTimeStamp())
+                .acknowledgement(dataAcknowledgement)
+                .resp(responseRequestId)
+                .build();
+      } else {
+        errorResponse.setMessage("HIP -> Mismatch of careContext");
+        errorResponse.setCode(GatewayConstants.ERROR_CODE);
+        log.error("OnInit body -> making error body since careContexts are not matched");
+        hipOnNotifyRequest =
+            HIPOnNotifyRequest.builder()
+                .requestId(UUID.randomUUID().toString())
+                .timestamp(Utils.getCurrentTimeStamp())
+                .error(errorResponse)
+                .resp(responseRequestId)
+                .build();
+      }
+      try {
+        log.info(hipOnNotifyRequest.toString());
+        ResponseEntity<GatewayGenericResponse> response =
+            requestManager.fetchResponseFromGateway(consentHipOnNotifyPath, hipOnNotifyRequest);
+        log.debug(consentHipOnNotifyPath + " : consentOnNotify: " + response.getStatusCode());
+        if (response.getStatusCode() == HttpStatus.ACCEPTED) {
+          requestLogService.dataTransferNotify(
```

Comment:
this is not data transfer again.. this should be HIP_ON_NOTIFY_SUCCESS and HIP_ON_NOTIFY_ERROR